lib/bulk_insert/worker.rb

```diff
@@ -70,38 +74,51 @@ def after_save(&block)
 
     def save!
       if pending?
-        sql = "INSERT #{@ignore} INTO #{@table_name} (#{@column_names}) VALUES "
-        @now = Time.now
-
         @before_save_callback.(@set) if @before_save_callback
+        @connection.execute(compose_insert_query) if compose_insert_query
```

Comment:
This will generate the query twice on each save. Maybe use `Object#tap` here?

```ruby
compose_insert_query.tap { |query| @connection.execute(query) if query }
```

Feel free to tackle it however you like, though, just so long as we only generate the query once per save.